benchmarks/suites/polybench/heat-3d/heat-3d.omp_target.cxx

```diff
@@ -35,9 +35,16 @@ static void kernel(pbsize_t tsteps, pbsize_t n,
 
 
 void run(State &state, pbsize_t pbsize) {
-  pbsize_t tsteps = 1; // 500
-  pbsize_t n = pbsize; // 120
-
+  pbsize_t tsteps = pbsize; // 500
+  if (pbsize <= 40) {
+      pbsize_t n = pbsize / 2;
```

Comment:
1. This does not work, `n` is declared inside the scope and not accessible outside.
2. [serious] It MUST be consistent with all heat-3d PPM sources, or verify will fail
3. If we move to the model, please apply it to all Polybench benchmarks
4. Since it is unrelated, consider doing it in a different patch
5. Consider [linear interpolation](https://en.wikipedia.org/wiki/Linear_interpolation#Linear_interpolation_as_approximation)